rasa/importers/importer.py

```diff
@@ -0,0 +1,258 @@
+        )
+
+    async def get_story_data(
+        self,
+        interpreter: "NaturalLanguageInterpreter" = RegexInterpreter(),
+        template_variables: Optional[Dict] = None,
+        use_e2e: bool = False,
+        exclusion_percentage: Optional[int] = None,
+    ) -> StoryGraph:
+        story_graphs = []
+        # Do this in a loop because Python 3.5 does not support async comprehensions
+        for importer in self._importers:
+            graph = await importer.get_story_data(
+                interpreter, template_variables, use_e2e, exclusion_percentage
+            )
+            story_graphs.append(graph)
+
+        return reduce(
+            lambda merged, other: merged.merge(other), story_graphs, StoryGraph([])
+        )
+
+    async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
+        nlu_datas = []
+        # Do this in a loop because Python 3.5 does not support async comprehensions
+        for importer in self._importers:
+            nlu_data = await importer.get_nlu_data(language)
+            nlu_datas.append(nlu_data)
+        training_data = reduce(
+            lambda merged, other: merged.merge(other), nlu_datas, TrainingData()
+        )
+        return training_data
+
+
+class SimpleFileImporter(TrainingFileImporter):
+    """Default `TrainingFileImporter` implementation."""
+
+    def __init__(
+        self,
+        config_file: Optional[Text] = None,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[Union[List[Text], Text]] = None,
+    ):
+        if config_file and os.path.exists(config_file):
+            self.config = io_utils.read_config_file(config_file)
+        else:
+            self.config = {}
+
+        self._domain_path = domain_path
+
+        if training_data_paths:
+            self._story_files, self._nlu_files = data.get_core_nlu_files(
+                training_data_paths
+            )
+
+    async def get_config(self) -> Dict:
+        return self.config
+
+    async def get_story_data(
+        self,
+        interpreter: "NaturalLanguageInterpreter" = RegexInterpreter(),
+        template_variables: Optional[Dict] = None,
+        use_e2e: bool = False,
+        exclusion_percentage: Optional[int] = None,
+    ) -> StoryGraph:
+
+        story_steps = await StoryFileReader.read_from_files(
+            self._story_files,
+            await self.get_domain(),
+            interpreter,
+            template_variables,
+            use_e2e,
+            exclusion_percentage,
+        )
+        return StoryGraph(story_steps)
+
+    async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
+        return utils.training_data_from_paths(self._nlu_files, language)
+
+    async def get_domain(self) -> Domain:
+        domain = Domain.empty()
+        try:
+            domain = Domain.load(self._domain_path)
+            domain.check_missing_templates()
+        except Exception:
```

Comment:
Why `Exception` and not `InvalidDomain`?

Reply:
👍 had to change it in `Domain` so that it also gives an `InvalidDomain` in case it's not found